packages/victory-core/src/victory-util/wrapper.js

```diff
@@ -204,6 +208,50 @@ export default {
     const { polar, startAngle, endAngle, categories, minDomain, maxDomain } = props;
     const parentProps = { polar, startAngle, endAngle, categories, minDomain, maxDomain };
     let stack = 0;
+    const children = childComponents
+      ? childComponents.slice(0)
+      : React.Children.toArray(props.children);
+
+    const someChildrenAreHistograms = children.some((child) => {
+      return child.type && child.type.role === "histogram";
+    });
+
+    const allChildrenAreHistograms =
+      children.length &&
+      children.every((child) => {
+        return child.type && child.type.role === "histogram";
+      });
+
+    if (someChildrenAreHistograms && !allChildrenAreHistograms) {
+      Log.warn(
+        "VictoryHistogram only supports being stacked with other VictoryHistogram components. Check to make sure that you are only passing VictoryHistogram components to VictoryStack"
+      );
+    }
+
+    // if we are stacking histograms, we need to generate explicit bins
+    // or else each histogram may end up having different bins
+    if (allChildrenAreHistograms) {
+      let childBins = props.bins || childComponents[0].props.bins;
```

Comment:
I think it would be more readable to pull all the bin logic into a separate function